File: java/universalChat/src/main/java/com/github/fernthedev/universal/PacketHandler.java
package com.github.fernthedev.universal;

import com.github.fernthedev.packets.*;
import com.google.protobuf.GeneratedMessageV3;

import java.util.ArrayList;
import java.util.List;

public class PacketHandler {

    public static List<GeneratedMessageV3> getPacketInstances() {
        List<GeneratedMessageV3> packets = new ArrayList<>();

        packets.add(AutoCompletePacket.getDefaultInstance());
        packets.add(ConnectedPacket.getDefaultInstance());
        packets.add(IllegalConnectionPacket.getDefaultInstance());
        packets.add(MessagePacket.getDefaultInstance());
        packets.add(RequestInfoPacket.getDefaultInstance());
        packets.add(SelfMessagePacket.getDefaultInstance());

        return packets;
    }

}

File: flutter/lib/backend/packethandler.dart
import 'package:chatclientflutter/packets/Packet.pb.dart';
import 'package:protobuf/protobuf.dart';
import 'dart:io';

class PacketHandler {

  static List<GeneratedMessage> packets = List<GeneratedMessage>();

  PacketHandler() {
    packets.clear();
    print("Listing packets");
    packets.add(AutoCompletePacket.getDefault());
    packets.add(ConnectedPacket.getDefault());
    packets.add(IllegalConnectionPacket.getDefault());
    packets.add(SelfMessagePacket.getDefault());
    packets.add(MessagePacket.getDefault());
    packets.add(RequestInfoPacket.getDefault());


  }


  static GeneratedMessage parseFromBytes(List<int> bytes) {
    GeneratedMessage generatedMessage;

    for(int i = 0; i < packets.length; i++) {
      GeneratedMessage checkMessage = packets[i];

      try {
        generatedMessage = checkMessage.createEmptyInstance();

        generatedMessage.mergeFromBuffer(bytes);

        _validateMessage(generatedMessage);

        print("Checking if $generatedMessage is message");

        if (!generatedMessage.isInitialized())
          continue;
        else
          break;
      } catch(e) {
        String check = checkMessage.runtimeType.toString();
        print("Error while attempting to parse $check");
        int length = packets.length;
        print("Int $i length: $length");
        print("Error: $e");

        if(i == packets.length -1) {
          print("Could not parse packet");
          generatedMessage = null;
          rethrow;
        }
        continue;
      }
    }

    String runtimeType = generatedMessage.runtimeType.toString();
    print("It is return $runtimeType");
    return generatedMessage;
  }

  static _validateMessage(GeneratedMessage generatedMessage) {
    generatedMessage.check();
  }
}
